### ISSUE TITLE
MCScan catalog error

### ISSUE THREAD

**Issue by SAMANDOLL** [tanghaibao/jcvi#76]
@vivekkrish @tanghaibao @Jingping  HI,I want use mcscan for synteny analysis.i try with your data example working good but when i will try for other crops error is coming bed.file.But bed.file have no problem.i search error gene name in bed.file, no issue in bed file but compare catalog is not working.can you fix  this problem for me.

17:31:06 [base] lastdb oryza oryza.cds
17:31:40 [base] lastal -u 0 -P 4 -i3G -f BlastTab oryza grape.cds >grape.oryza.last
17:32:45 [synteny] Assuming --qbed=grape.bed --sbed=oryza.bed
17:32:45 [base] Load file `grape.bed`
17:32:45 [base] Load file `oryza.bed`
17:32:46 [blastfilter] Load BLAST file `grape.oryza.last` (total 357447 lines)
17:32:46 [base] Load file `grape.oryza.last`
17:32:49 [blastfilter] LOC_Os05g07050 not in oryza.bed
17:32:49 [blastfilter] LOC_Os11g01380 not in oryza.bed
17:32:49 [blastfilter] LOC_Os12g01390 not in oryza.bed
17:32:49 [blastfilter] LOC_Os11g01380 not in oryza.bed
17:32:49 [blastfilter] LOC_Os12g01390 not in oryza.bed
17:32:49 [blastfilter] LOC_Os02g47970 not in oryza.bed
17:32:49 [blastfilter] LOC_Os07g46460 not in oryza.bed
17:32:49 [blastfilter] LOC_Os01g48960 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g48140 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g48140 not in oryza.bed
17:32:49 [blastfilter] LOC_Os05g48200 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g03610 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g03610 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g03610 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g03610 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g03610 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g03610 not in oryza.bed
17:32:49 [blastfilter] LOC_Os06g29350 not in oryza.bed
17:32:49 [blastfilter] LOC_Os06g07080 not in oryza.bed
17:32:49 [blastfilter] LOC_Os06g29350 not in oryza.bed
17:32:49 [blastfilter] LOC_Os12g30540 not in oryza.bed
17:32:49 [blastfilter] LOC_Os12g30540 not in oryza.bed
17:32:49 [blastfilter] LOC_Os01g46510 not in oryza.bed
17:32:49 [blastfilter] LOC_Os01g46510 not in oryza.bed
17:32:49 [blastfilter] LOC_Os12g30540 not in oryza.bed
17:32:49 [blastfilter] LOC_Os06g08380 not in oryza.bed
17:32:49 [blastfilter] LOC_Os11g08330 not in oryza.bed
17:32:49 [blastfilter] LOC_Os07g37750 not in oryza.bed
17:32:49 [blastfilter] LOC_Os02g57190 not in oryza.bed
17:32:49 [blastfilter] LOC_Os12g44150 not in oryza.bed
17:32:49 [blastfilter] LOC_Os03g48310 not in oryza.bed
17:32:49 [blastfilter] LOC_Os04g56160 not in oryza.bed

**Comment by vtartaglio** [comment]
@tanghaibao Hi sorry I know this is an old thread, but I am getting the exact same error. The geneIDs in my cds, gff, and species1.species2.last file all match and are correct. All my geneIDs end with an isoform number e.g. Brahy.D03G0373200.2. However, at the stage described by SAMANDOLL above the isoform number is apparently being dropped, and I am getting the error "10:40:09 [blastfilter] Brahy.D03G0373200 not in BhD.bed". Is there a known solution to this? Should I edit my input files to contain only the primary isoform? Thanks so much.

**Comment by tanghaibao** [comment]
@vtartaglio 

This is a common issue and a source of confusion for people who are new to the package. 

By default, the script will drop isoforms (by default, e.g. `At5g006540.1` becomes `At5g06540`), as a result, your `.bed` file needs to contain `At5g06540` to match those ids.

Alternatively, you could do `--no_strip_names` when you run `ortholog` and the isoform number will be retained. But in this case, your `.bed` file needs to contain `At5g006540.1` to match the ids.

```console
python -m jcvi.compara.catalog ortholog --no_strip_names
```

It would be useful to know that the `.bed` file converter has an option of `--strip_names` as well.

```console
python -m jcvi.formats.gff bed athaliana.gff -o athaliana.bed
```

When running this command, notice that due to the differences in `.gff` files, you may want to specify `--key` and `--type`. The important thing is to match up the ids in `.fasta` and `.bed` (and pay attention to the isoform dropping). Notice how these commands are called in the [wiki](https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version)).

Haibao

**Comment by vtartaglio** [comment]
Okay thank you so much for your prompt response! Even with --no_strip_names I was still getting odd results: I got only 3 anchors, strangely. There must have been some incompatibility between the isoforms present in my Phytozome cds and gff files. Maybe the gff includes non-coding RNAs or something... Anyway, I just reformatted my inputs to include the primary isoform only, and that fixed the problem. Sorry for my slow response, thanks again!
